### ISSUE TITLE
Reporter - Fails to generate docx, html and xlsx reports

### ISSUE THREAD

**Issue by pkievit-copado** [damies13/rfswarm#144]
**Describe the bug**
Reporter - Fail to generate docx, html and xlsx reports

**To Reproduce**

rfswarm-reporter -n -i '/Users/patrick/Library/Application Support/Copado/QConnect/execution/performance-testing/files/configuration/RFSwarmReporter.ini' --debug 1 -d '/Users/patrick/Library/Application Support/Copado/QConnect/execution/performance-testing/20230424_105440_qconnect-load-test/20230424_105440_qconnect-load-test.db' -t '/Users/patrick/Library/Application Support/Copado/QConnect/execution/performance-testing/files/reporting/default.template' --html --xlsx --docx

**Expected behavior**
Reports formats should be generated.

**Desktop (please complete the following information):**
- OS: Mac OS Ventura
- Version version 1.1.0 latest using standalone_python version 3.9.7

**Log**

`rfswarm-reporter -n -i '/Users/patrick/Library/Application Support/Copado/QConnect/execution/performance-testing/files/configuration/RFSwarmReporter.ini' --debug 1 -d '/Users/patrick/Library/Application Support/Copado/QConnect/execution/performance-testing/20230424_105440_qconnect-load-test/20230424_105440_qconnect-load-test.db' -t '/Users/patrick/Library/Application Support/Copado/QConnect/execution/performance-testing/files/reporting/default.template' --html --xlsx --docx

Robot Framework Swarm: Reporter 

	Version 1.1.0 

self.args.ini:  /Users/patrick/Library/Application Support/Copado/QConnect/execution/performance-testing/files/configuration/RFSwarmReporter.ini 

	Configuration File:  /Users/patrick/Library/Application Support/Copado/QConnect/execution/performance-testing/files/configuration/RFSwarmReporter.ini 

Generating XHTML Report 

Generating Word Report 

Exception in thread Thread-1:

Traceback (most recent call last):

  File "/Applications/QConnect.app/Contents/Resources/standalone_python/install/lib/python3.9/threading.py", line 973, in _bootstrap_inner

Generating Excel Report 

Exception in thread Thread-3:

Traceback (most recent call last):

  File "/Applications/QConnect.app/Contents/Resources/standalone_python/install/lib/python3.9/threading.py", line 973, in _bootstrap_inner

    self.run()

  File "/Applications/QConnect.app/Contents/Resources/standalone_python/install/lib/python3.9/threading.py", line 910, in run

    self._target(*self._args, **self._kwargs)

  File "/Applications/QConnect.app/Contents/Resources/standalone_python/install/lib/python3.9/site-packages/rfswarm_reporter/rfswarm-reporter", line 2524, in export_xhtml

    self.run()

  File "/Applications/QConnect.app/Contents/Resources/standalone_python/install/lib/python3.9/threading.py", line 910, in run

    fontname = base.rs_setting_get_font()

    self._target(*self._args, **self._kwargs)

  File "/Applications/QConnect.app/Contents/Resources/standalone_python/install/lib/python3.9/site-packages/rfswarm_reporter/rfswarm-reporter", line 2621, in export_excel

  File "/Applications/QConnect.app/Contents/Resources/standalone_python/install/lib/python3.9/site-packages/rfswarm_reporter/rfswarm-reporter", line 533, in rs_setting_get_font

    fontlst = list(tkFont.families())

  File "/Applications/QConnect.app/Contents/Resources/standalone_python/install/lib/python3.9/tkinter/font.py", line 183, in families

    root = tkinter._get_default_root('use font.families()')

  File "/Applications/QConnect.app/Contents/Resources/standalone_python/install/lib/python3.9/tkinter/__init__.py", line 297, in _get_default_root

    raise RuntimeError(f"Too early to {what}: no default root window")

RuntimeError: Too early to use font.families(): no default root window

    self.xlsx_configure_style()

  File "/Applications/QConnect.app/Contents/Resources/standalone_python/install/lib/python3.9/site-packages/rfswarm_reporter/rfswarm-reporter", line 3772, in xlsx_configure_style

    fontname = base.rs_setting_get_font()

  File "/Applications/QConnect.app/Contents/Resources/standalone_python/install/lib/python3.9/site-packages/rfswarm_reporter/rfswarm-reporter", line 533, in rs_setting_get_font

    fontlst = list(tkFont.families())

  File "/Applications/QConnect.app/Contents/Resources/standalone_python/install/lib/python3.9/tkinter/font.py", line 183, in families

    root = tkinter._get_default_root('use font.families()')

  File "/Applications/QConnect.app/Contents/Resources/standalone_python/install/lib/python3.9/tkinter/__init__.py", line 297, in _get_default_root

    raise RuntimeError(f"Too early to {what}: no default root window")

RuntimeError: Too early to use font.families(): no default root window

Exception in thread Thread-2:

Traceback (most recent call last):

  File "/Applications/QConnect.app/Contents/Resources/standalone_python/install/lib/python3.9/threading.py", line 973, in _bootstrap_inner

    self.run()

  File "/Applications/QConnect.app/Contents/Resources/standalone_python/install/lib/python3.9/threading.py", line 910, in run

    self._target(*self._args, **self._kwargs)

  File "/Applications/QConnect.app/Contents/Resources/standalone_python/install/lib/python3.9/site-packages/rfswarm_reporter/rfswarm-reporter", line 2585, in export_word

    self.docx_configure_style()

  File "/Applications/QConnect.app/Contents/Resources/standalone_python/install/lib/python3.9/site-packages/rfswarm_reporter/rfswarm-reporter", line 3158, in docx_configure_style

    fontname = base.rs_setting_get_font()

  File "/Applications/QConnect.app/Contents/Resources/standalone_python/install/lib/python3.9/site-packages/rfswarm_reporter/rfswarm-reporter", line 533, in rs_setting_get_font

    fontlst = list(tkFont.families())

  File "/Applications/QConnect.app/Contents/Resources/standalone_python/install/lib/python3.9/tkinter/font.py", line 183, in families

    root = tkinter._get_default_root('use font.families()')

  File "/Applications/QConnect.app/Contents/Resources/standalone_python/install/lib/python3.9/tkinter/__init__.py", line 297, in _get_default_root

    raise RuntimeError(f"Too early to {what}: no default root window")

RuntimeError: Too early to use font.families(): no default root window

patrick@MacBook-Pro-van-Patrick ~ % `

**Comment by damies13** [comment]
I can confirm it's related to this issue you mentioned in Discord

https://stackoverflow.com/questions/69792938/tkinter-fonts-in-a-seprate-file

When using -n there is no tkinter window (nor should there be), but rs_setting_get_font function is using tkinter to load the operating system's font list as I was trying to prevent another bug in the GUI where it would try load the preview with a font chosen in the template that didn't exist in the OS. Here's the offending code:

```python
	def rs_setting_get_font(self):
		fontlst = list(tkFont.families())
		base.debugmsg(9, "fontlst", fontlst)
		value = self.rs_setting_get('font')
		if value not in fontlst:
			value = None
		if value is None:
			# Verdana, Tahoma, Arial, Helvetica, sans-serif
			fontorder = ['Helvetica', 'Verdana', 'Tahoma', 'Arial', 'FreeSans']
			for fnt in fontorder:
				if fnt in fontlst:
					return fnt
			for fnt in fontlst:
				if 'Sans' in fnt or 'sans' in fnt:
					return fnt
			return 'sans-serif'
		else:
			return value

```

My initial thought is to not load the font list first, but rather read the value form the template and if running with -n just return the font name without any checks, then only if in GUI mode do the checks.

My concern here is how the docx and xlsx generation will work, will they error is the font is not found? 
The HTML generation will work and the browsers will handle the situation if the font is not there.

I guess I need to try it out and see what happens

**Comment by damies13** [comment]
- [x] Regression test added
- [x] Fix Applied 

This is the first issue with a regression test as I added the ability to run regression tests on Windows, Mac and Linux (Ubuntu) on several versions of python each as part of V1.1.0. Now there is a regression test for this we should be able to catch it if this issue occurs again.

**Comment by damies13** [comment]
So far this bug report has been very productive, in addition to the original issue it's also found:
- where I missed an change for Issue #136
- The same issue as Issue #97, even though that issue was closed before Reporter was created, I should have copied the fix across from Manager and I missed it, so this was a good catch before it impacted anyone. This was the Windows specific issue, MacOS and Linux use UTF-8 by default, Windows uses CP-1252

Now all builds are passing except [Manager (macos-latest, 3.10)](https://github.com/damies13/rfswarm/actions/runs/4816538642/jobs/8576267228#logs), which I'm going to ignore for now as:
- I didn't make any changes to the Manager code 
- It's a known issue with the github build servers (https://github.com/actions/setup-python/issues/649) and should hopefully be fixed soon

**Comment by damies13** [comment]
Merged into release v1.1.1 branch
